crc32.c

```diff
@@ -740,7 +733,14 @@ unsigned long ZEXPORT crc32_z(crc, buf, len)
     z_size_t len;
 {
     /* Return initial CRC, if requested. */
-    if (buf == Z_NULL) return 0;
+    if (buf == Z_NULL) {
+	/* Assume user is calling 'crc32(0, NULL, 0)', so we cache CPU features
+	 * detection early (and infrequently) on.
+	 */
+        if (!len)
+            cpu_check_features();
```

Comment:
An explanation here: in Chromium code base, we have client code (e.g. network code) that may rely on using zlib's crc32.

There is no guarantee that the client code is performing compression/decompression of data, so we have to cover that corner case by ensuring that we perform CPU features detection in a scenario where the user only rely on calling crc32() (e.g. first to get an initial valid crc32 value and next with a real data vector).